### PR TITLE
[zh] translate user-namespace

### PR DESCRIPTION
**Pull request by 0xff-dev**
[zh] translate user-namespace

### CODE REVIEW

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+* containerd：计划在1.7版本中支持。更多细节请参见 containerd 问题 [#7063][containerd-userns-issue]。
+
+目前 [cri-dockerd 没有计划][CRI-dockerd-issue]支持此功能。
+
+[CRI-dockerd-issue]: https://github.com/Mirantis/cri-dockerd/issues/74
+[containerd-userns-issue]: https://github.com/containerd/containerd/issues/7063
+
+<!-- ## Introduction -->
+## 介绍
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:74]:
```suggestion
## 介绍   {#introduction}
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+* CRI-O: v1.25 has support for user namespaces.
+
+* containerd: support is planned for the 1.7 release. See containerd
+  issue [#7063][containerd-userns-issue] for more details.
+
+Support for this in [cri-dockerd is not planned][CRI-dockerd-issue] yet.
+-->
+
+这只是一个Linux的功能。此外，需要在 {{< glossary_tooltip text="容器运行时" term_id="container-runtime" >}} 提供支持，
+以便在 Kubernetes 无状态 Pods 中使用这一功能：
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:62]:
```suggestion
这是一个仅 Linux 适用的功能。
此外只有{{< glossary_tooltip text="容器运行时" term_id="container-runtime" >}}提供支持，
才能在 Kubernetes 无状态 Pod 中使用这一功能：
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
+---
+title: 用户命名空间
+reviewers:
+content_type: concept
+weight: 160
+min-kubernetes-server-version: v1.25
+---
+
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:8]:
```suggestion
<!--
title: User Namespaces
reviewers:
content_type: concept
weight: 160
min-kubernetes-server-version: v1.25
-->
```

**Review by windsonsea — COMMENTED** [review]
Thanks for your contributions. Here are some tips:

- Add an anchor to heading 2 and 3 if necessary
- Provide a space between Chinese and English characters

More details see [中文本地化样式指南](https://kubernetes.io/zh-cn/docs/contribute/localization_zh/).

**Comment by 0xff-dev** [comment]
> Thanks for your contributions. Here are some tips:
> 
> * Add an anchor to heading 2 and 3 if necessary
> * Provide a space between Chinese and English characters
> 
> More details see [中文本地化样式指南](https://kubernetes.io/zh-cn/docs/contribute/localization_zh/).

Thank you for the review.

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+- `CAP_SYS_ADMIN` is limited to the pod's user namespace and invalid outside
+of it.
+-->
+此外，由于每个 Pod 上的用户将被映射到主机中不同的非重叠用户，
+他们对其他 Pod 的操作也是有限的。
+
+授予一个 Pod 的能力也被限制在 Pod 的用户命名空间内，
+并且在它之外大多无效，有些甚至完全无效。这里有两个例子：
+
+- `CAP_SYS_MODULE` 授予一个使用用户名字空间的 Pod 没有任何效果，这个 Pod 不能加载内核模块。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:193]:
```suggestion
- `CAP_SYS_MODULE` 若被授予一个使用用户名字空间的 Pod 则没有任何效果，这个 Pod 不能加载内核模块。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+- `CAP_SYS_ADMIN` 只限于 Pod 的用户名字空间，在它之外无效。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:194]:
```suggestion
- `CAP_SYS_ADMIN` 只限于 Pod 所在的用户名字空间，在该名字空间之外无效。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+<!--
+Without using a user namespace a container running as root, in the case of a
+container breakout, has root privileges on the node. And if some capability were
+granted to the container, the capabilities are valid on the host too. None of
+this is true when we use user namespaces.
+
+If you want to know more details about what changes when user namespaces are in
+use, see `man 7 user_namespaces`.
+-->
+在不使用用户命名空间的情况下，作为 root 运行的容器，在容器突破的情况下，在节点上有 root 权限。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:205]:
```suggestion
在不使用用户命名空间的情况下，以 root 账号运行的容器，在容器逃逸时，在节点上有 root 权限。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+而且如果某些能力被授予了容器，这些能力在主机上也是有效的。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:206]:
```suggestion
而且如果某些权能被授予了某容器，这些权能在宿主机上也是有效的。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+当我们使用用户命名空间时，这些都不是真的。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:207]:
```suggestion
当我们使用用户命名空间时，这些都不再成立。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+
+如果你想知道更多关于使用用户命名空间时的变化细节，请参见 `man 7 user_namespaces`。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:209]:
```suggestion
如果你想知道关于使用用户命名空间时的更多变化细节，请参见 `man 7 user_namespaces`。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+
+
+<!--## Set up a node to support user namespaces-->
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:212]:
```suggestion
<!--
## Set up a node to support user namespaces
-->
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+files in the hosts. If the UIDs/GIDs of the pod and the host don't overlap, it
+is limited what a pod would be able to do: the pod UID/GID won't match the
+host's file owner/group.
+-->
+建议主机的文件和主机的进程使用 0-65535 范围内的 UID/GID。
+
+kubelet 会把高于这个范围的 UID/GID 分配给 Pod。
+因此，为了保证尽可能多的隔离，主机的文件和主机的进程所使用的 UID/GID 应该在 0-65535 范围内。
+
+请注意，这个建议对减轻 [CVE-2021-25741][CVE-2021-25741] 等 CVE 的影响很重要，
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:234]:
```suggestion
请注意，这个建议对减轻 [CVE-2021-25741][CVE-2021-25741] 等 CVE 的影响很重要；
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
+在这些 CVE 中，Pod 有可能读取主机中的任意文件。
+如果 Pod 和主机的 UID/GID 不重叠，Pod 能够做的事情就会受到限制：
+Pod的 UID/GID 不会与主机的文件所有者/组相匹配。
+
+[CVE-2021-25741]: https://github.com/kubernetes/kubernetes/issues/104980
+
+<!-- # Limitations -->
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:241]:
```suggestion
<!--
# Limitations
-->
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 13]
```diff
@@ -0,0 +1,293 @@
...
+<!--
+When using a user namespace for the pod, it is disallowed to use other host
+namespaces. In particular, if you set `hostUsers: false` then you are not
+allowed to set any of:
+
+ * `hostNetwork: true`
+ * `hostIPC: true`
+ * `hostPID: true`
+-->
+当为 Pod 使用一个用户命名空间时，不允许使用其他主机命名空间。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:252]:
```suggestion
当 Pod 使用用户命名空间时，不允许 Pod 使用其他主机命名空间。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 14]
```diff
@@ -0,0 +1,293 @@
...
+特别是，如果你设置 `hostUsers: false`，那么你不允许设置任何。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:253]:
```suggestion
特别是，如果你设置了 `hostUsers: false`，那么你就不可以设置如下属性：
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 15]
```diff
@@ -0,0 +1,293 @@
...
+The pod is allowed to use no volumes at all or, if using volumes, only these
+volume types are allowed:
+
+ * configmap
+ * secret
+ * projected
+ * downwardAPI
+ * emptyDir
+-->
+允许 Pod 完全不使用卷，如果使用卷，只允许使用这些卷的类型。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:269]:
```suggestion
Pod 完全不使用卷是被允许的；如果使用卷，只允许使用以下卷类型：
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 16]
```diff
@@ -0,0 +1,293 @@
...
+
+ * configmap
+ * secret
+ * projected
+ * downwardAPI
+ * emptyDir
+
+<!-- To guarantee that the pod can read the files of such volumes, volumes are
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:277]:
```suggestion
<!--
To guarantee that the pod can read the files of such volumes, volumes are
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 17]
```diff
@@ -0,0 +1,293 @@
...
+created as if you specified `.spec.securityContext.fsGroup` as `0` for the Pod.
+If it is specified to a different value, this other value will of course be
+honored instead.
+
+As a by-product of this, folders and files for these volumes will have
+permissions for the group, even if `defaultMode` or `mode` to specific items of
+the volumes were specified without permissions to groups. For example, it is not
+possible to mount these volumes in a way that its files have permissions only
+for the owner. -->
+
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:287]:
```suggestion
for the owner.
-->
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 18]
```diff
@@ -0,0 +1,293 @@
...
+为了保证 Pod 可以读取这些卷的文件，卷的创建就像你为 Pod 指定 `.spec.securityContext.fsGroup` 为 `0` 一样。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:288]:
```suggestion
为了保证 Pod 可以读取这些卷中的文件，卷的创建操作就像你为 Pod 指定了 `.spec.securityContext.fsGroup` 为 `0` 一样。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 19]
```diff
@@ -0,0 +1,293 @@
...
+如果它被指定为一个不同的值，这个其他的值当然会被取代。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:289]:
```suggestion
如果该属性被设定为不同值，那么这个不同值当然也会被使用。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 20]
```diff
@@ -0,0 +1,293 @@
...
+
+作为一个副产品，这些卷的文件夹和文件将具有组的权限，
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:291]:
```suggestion
作为一个副产品，这些卷的文件夹和文件将具有所给组的权限，
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 21]
```diff
@@ -0,0 +1,293 @@
...
+即使 `defaultMode` 或卷的特定项目的 `mode` 被指定为没有组的权限。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:292]:
```suggestion
即使 `defaultMode` 或 volumes 的特定项目的 `mode` 被指定为没有组的权限。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 22]
```diff
@@ -0,0 +1,293 @@
...
+例如，不可能在挂载这些卷时使其文件只对所有者有权限。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:293]:
```suggestion
例如，不可以在挂载这些卷时使其文件只允许所有者访问。
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 23]
```diff
@@ -0,0 +1,298 @@
...
+weight: 160
+min-kubernetes-server-version: v1.25
+-->
+---
+title: 用户命名空间
+reviewers:
+content_type: concept
+weight: 160
+min-kubernetes-server-version: v1.25
+---
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:14]:
```suggestion
---
title: 用户命名空间
content_type: concept
weight: 160
min-kubernetes-server-version: v1.25
---
<!--
title: User Namespaces
reviewers:
content_type: concept
weight: 160
min-kubernetes-server-version: v1.25
-->
```

File: content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md [hunk 24]
```diff
@@ -0,0 +1,297 @@
...
+of it.
+-->
+此外，由于每个 Pod 上的用户将被映射到主机中不同的非重叠用户，
+他们对其他 Pod 可以执行的操作也是有限的。
+
+授予一个 Pod 的权能也被限制在 Pod 的用户命名空间内，
+并且在这一名字空间之外大多无效，有些甚至完全无效。这里有两个例子：
+
+ - `CAP_SYS_MODULE` 若被授予一个使用用户名字空间的 Pod 则没有任何效果，这个 Pod 不能加载内核模块。
+ - `CAP_SYS_ADMIN` 只限于 Pod 所在的用户名字空间，在该名字空间之外无效。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/user-namespaces.md:194]:
```suggestion
并且在这一命名空间之外大多无效，有些甚至完全无效。这里有两个例子：

 - `CAP_SYS_MODULE` 若被授予一个使用用户命名空间的 Pod 则没有任何效果，这个 Pod 不能加载内核模块。
 - `CAP_SYS_ADMIN` 只限于 Pod 所在的用户命名空间，在该命名空间之外无效。
```

**Review by windsonsea — COMMENTED** [review]
Pick a nit:

- 34 `命名空间`
- 4 `名字空间`

It's better to keep them consistent for the first translation.

**Comment by windsonsea** [comment]
Fix #37624

**Comment by tengqm** [comment]
/approve

It is okay to merge once the nits suggested are fixed.

**Comment by 0xff-dev** [comment]
@tengqm @windsonsea thanks for your review.

**Pull request merged**